metis-core/metis-core-service/src/main/java/eu/europeana/metis/core/execution/WorkflowExecutor.java

```diff
@@ -408,36 +406,21 @@ private void periodicCheckingLoop(long sleepTime, AbstractExecutablePlugin plugi
             workflowExecution.getId(), plugin.getPluginType()), e);
         currentThread().interrupt();
         return;
-      } catch (ExternalTaskException e) {
-        final Throwable cause = ExternalRequestUtil.getRootCause(e);
-        // TODO: 14/05/2021 We might want to remove this or find a better way to handle the shutdown of metis-core because, it seems that this occurred during an execution of a plugin and therefore the plugin went out of the loop.
-        // The next plugin was tried but was marked FAILED because the previous plugin didn't finish and it automatically marked the previous plugin CANCELLED
-        // which shouldn't happen.(The plugin was still in progress in ecloud)
-//        if (cause instanceof IllegalStateException) {
-//          //If the application has a forcible shutdown we might experience the JerseyClient in DpsClient
-//          // internally closing down without our consent even if we would have a synchronized close
-//          // implemented. This catch gives us a little more assurance of avoiding an execution
-//          // being marked as failed.
-//          LOGGER.warn("Application is probably shutting down at the moment and dpsClient has "
-//              + "closed without our consent, so we are ignoring this exception.", e);
-//          return;
-//        }
+      }catch (AccessDeniedOrObjectDoesNotExistException e){
```

Comment:
1. We talked about it in our team and decided to not split AccessDeniedOrObjectDoesNotExistException.
Reasons:
-the user who does not have right to the object should not be able to check if the object exists. So both situations are indistinguishable. On this API level.
-we need to change eCloud api. 
-it is technically more difficult, we check acl now by annotations, and we need to check also if the object itself exists after it.

So, if you want to do the change anyway. Maybe lets meet in greater team together to discuss it.

2. I could not revert the whole code cause here is main change - Metis should not fail on any exception but the AccessDeniedOrObjectDoesNotExistException.
So I only changed the code to not catch AccessDeniedOrObjectDoesNotExistException directly but catch it wrapped by UnrecoverableExternalTaskException.

Reply:
* I pointed out the split `AccessDeniedOrObjectDoesNotExistException` because is not a good practice (Clean Code) at least in the naming (because indicates it does two things). We can leave it as it is 
* I understand now what you explain in point. no. 2 because I see `MessageBodyProviderNotFoundException` can still happen it comes from `org.glassfish.jersey.message.internal` dependency and the `IllegalStateException` comes from the `eu.europeana.cloud.service.mcs.persistent.uis.UISClientHandlerImpl` so in this cases we don't want metis to fail is that correct?